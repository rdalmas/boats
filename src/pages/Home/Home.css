.wrapper {
  height: 100vh;
  width: 100vw;
  background: url(../../img/boat.jpg) no-repeat bottom center;
  display: flex;
  align-items: center;
  justify-content: center;
}

.jumbotron {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  background: rgba(0, 0, 0, .5);
}

h1, .lead {
  color: #fff;
}

.container {
  padding: 75px 0;
}