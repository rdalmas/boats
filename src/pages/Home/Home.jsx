import React from 'react';
import { Link } from "react-router-dom";

import Wrapper from '../../styled-components/Wrapper';
import Panel from '../../styled-components/Panel';
import boatImg from "../../img/boat.jpg";
import { searchDefaults as def } from "../../constants";

export const Home = () => (
  <Wrapper background={boatImg}>
    <Panel>
      <div className="row justify-content-center text-center">
        <div className="col-md-12 col-lg-8">
          <h1 className="white-text">Your next boat trip starts here</h1>
          <h5 className="white-text">Find your next destination</h5>
          <Link className="btn btn-secondary btn-lg" to={{pathname: "/search", search: `?length=${def.minLength}&year=${def.minYear}`}} role="button">Search</Link>
        </div>
      </div>
    </Panel>
  </Wrapper>
);

export default Home;
