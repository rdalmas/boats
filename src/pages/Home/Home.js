import React from 'react';

import "./Home.css";

function Home() {
  return (
    <div className="wrapper">
      <div className="jumbotron jumbotron-fluid mb-0">
        <div className="container">
          <div className="row justify-content-center text-center">
          <div className="col-md-10 col-lg-6">
            <h1 className="display-5">Your next boat trip starts here</h1>
            <p className="lead">Find your next destination.</p>
            <p className="lead">
              <a className="btn btn-primary btn-lg" href="/search" role="button">Search</a>
            </p>
          </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Home;
