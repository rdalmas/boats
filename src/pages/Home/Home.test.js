import React from 'react';
import { shallow } from "enzyme";

import { Home } from './Home';

describe("Page - Home", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Home />);
  });
  it('should render the Home Page correctly', () => {   
    expect(component).toMatchSnapshot();
  });
});

