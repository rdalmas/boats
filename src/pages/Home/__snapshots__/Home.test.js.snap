// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - Home should render the Home Page correctly 1`] = `
<styled.div
  background="boat.jpg"
>
  <styled.div>
    <div
      className="row justify-content-center text-center"
    >
      <div
        className="col-md-12 col-lg-8"
      >
        <h1
          className="white-text"
        >
          Your next boat trip starts here
        </h1>
        <h5
          className="white-text"
        >
          Find your next destination
        </h5>
        <Link
          className="btn btn-secondary btn-lg"
          role="button"
          to={
            Object {
              "pathname": "/search",
              "search": "?length=0&year=1950",
            }
          }
        >
          Search
        </Link>
      </div>
    </div>
  </styled.div>
</styled.div>
`;
